### ISSUE TITLE
IncorrectCase doesn't catch patterns in function parameters

### ISSUE THREAD

**Issue by ArifRoktim** [rust-analyzer/rust-analyzer#6322]
Given:
```rust
pub fn func((INPUT_ONE, INPUT_TWO): (u8, u8)) {}
```
`INPUT_ONE`/`INPUT_TWO` aren't marked as having incorrect case.

**Comment by bjorn3** [comment]
```rust
const INPUT_ONE: () = ();
const INPUT_TWO: () = ();

pub fn func((INPUT_ONE, INPUT_TWO): ((), ())) {}
```

is valid rust and doesn't give a warning with rustc either. `INPUT_ONE` and `INPUT_TWO` match `()` here. This means that implementing this lint in this case will have to resolve identifiers, which can slow it down and will give spurious warnings on incomplete code elsewhere.

**Comment by ArifRoktim** [comment]
That's an interesting scenario that I didn't think of.
The only time you can create an irrefutable pattern with a `const` value is if that type has exactly 1 value (IIUC). Having a function argument that has only 1 possible value seems pointless to me. I don't think functions like this get written very often, if at all.

**Comment by ChayimFriedman2** [comment]
Works today.
